src/frontend/src/pages/HomePage/components/TimelineSection.tsx

```diff
@@ -1,8 +1,59 @@
-import { Typography } from '@mui/material';
+import { Grid, Typography } from '@mui/material';
+import * as React from 'react';
+import Timeline from '@mui/lab/Timeline';
+import TimelineItem from '@mui/lab/TimelineItem';
+import TimelineSeparator from '@mui/lab/TimelineSeparator';
+import TimelineConnector from '@mui/lab/TimelineConnector';
+import TimelineContent from '@mui/lab/TimelineContent';
+import TimelineDot from '@mui/lab/TimelineDot';
+import { useAllMilestones } from '../../../hooks/recruitment.hooks';
+import LoadingIndicator from '../../../components/LoadingIndicator';
+import ErrorPage from '../../ErrorPage';
+import dayjs from 'dayjs';
 
 const TimelineSection = () => {
-  // TODO: Ticket #2715
-  return <Typography>Timeline Section</Typography>;
+  const { isLoading, isError, error, data: mileStones } = useAllMilestones();
+
+  if (isLoading || !mileStones) return <LoadingIndicator />;
+  if (isError) return <ErrorPage error={error} message={error.message} />;
+
+  const isPastEvent = (date: Date) => dayjs(date).isBefore(dayjs());
```

Comment:
instead of using dayjs to check if it is past the event, can you just compare the date objects? I think that works. Like you can do return date < new date(); or smth like that

Reply:
this way you can get rid of the extra depedancy as well